pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,421 @@
+	installTest           = "install should succeed: overall"
+	installTestIdentifier = testIdentifier{installTestSuites, installTest}
+)
+
+// JobGetter gets related jobs for further analysis
+type JobGetter interface {
+	GetJobs(ctx context.Context) ([]jobrunaggregatorapi.JobRow, error)
+}
+
+type testCaseAnalyzerJobGetter struct {
+	platform       string
+	infrastructure string
+	network        string
+	jobGCSPrefixes *[]jobGCSPrefix
+	ciDataClient   jobrunaggregatorlib.CIDataClient
+}
+
+// GetJobs find all related jobs for the test case analyzer
+// For PR payload, this contains jobs correspond to the list of jobGCSPreix passed
+// For release-controller generated payload, this contains all jobs meeting selection criteria
+// from command args.
+func (s *testCaseAnalyzerJobGetter) GetJobs(ctx context.Context) ([]jobrunaggregatorapi.JobRow, error) {
+	jobs, err := s.ciDataClient.ListAllJobs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list all jobs: %w", err)
+	}
+
+	// if PR payload, only find the exact jobs
+	if len(*s.jobGCSPrefixes) > 0 {
+		jobNames := sets.String{}
+		for i := range *s.jobGCSPrefixes {
+			jobGCSPrefix := (*s.jobGCSPrefixes)[i]
+			jobNames.Insert(jobGCSPrefix.jobName)
+		}
+		jobs = s.filterJobsByNames(jobNames, jobs)
+	} else {
+		// Non PR payload, select by criteria
+		jobs = s.filterJobsForPayload(jobs)
+	}
+	return jobs, nil
+}
+
+func getJobInfrastructure(name string) string {
+	if strings.Contains(name, "upi") {
+		return "upi"
+	}
+	return "ipi"
+}
+
+func (s *testCaseAnalyzerJobGetter) filterJobsForPayload(allJobs []jobrunaggregatorapi.JobRow) []jobrunaggregatorapi.JobRow {
+	jobs := []jobrunaggregatorapi.JobRow{}
+	for i := range allJobs {
+		job := allJobs[i]
+		if (len(s.platform) != 0 && job.Platform != s.platform) ||
+			(len(s.network) != 0 && job.Network != s.network) ||
+			(len(s.infrastructure) != 0 && s.infrastructure != getJobInfrastructure(job.JobName)) {
+			continue
+		}
+		jobs = append(jobs, job)
+	}
+	return jobs
+}
+
+func (s *testCaseAnalyzerJobGetter) filterJobsByNames(jobNames sets.String, allJobs []jobrunaggregatorapi.JobRow) []jobrunaggregatorapi.JobRow {
+	ret := []jobrunaggregatorapi.JobRow{}
+	for i := range allJobs {
+		curr := allJobs[i]
+		if jobNames.Has(curr.JobName) {
+			ret = append(ret, curr)
+		}
+	}
+	return ret
+}
+
+// TestCaseChecker checks if a test passes certain criteria across all job runs
+type TestCaseChecker interface {
+	// CheckTestCase returns a test case based on whether a test has passed certain criteria across job runs
+	CheckTestCase(ctx context.Context, jobRuns []jobrunaggregatorapi.JobRunInfo) *junit.TestCase
+}
+
+type minimumRequiredPassesTestCaseChecker struct {
+	id                     testIdentifier
+	testNameSuffix         string
+	requiredNumberOfPasses int
+}
+
+type testStatus int
+
+const (
+	testSkipped testStatus = iota
+	testPassed
+	testFailed
+)
+
+func getTestStatus(id testIdentifier, testSuite *junit.TestSuite) testStatus {
```

Comment:
If you pass a testSuite containing multiple testcases, only the first matching hit is checked.  This case needs to produce an error.

Reply:
> If you pass a testSuite containing multiple testcases, only the first matching hit is checked. This case needs to produce an error.

ok, I think I see it.  It's an unfortunate naming issue in the testSuite versus ID.